### PR TITLE
Graph Tweaks

### PR DESCRIPTION
**Pull request by 3v0k4**
- [x] render empty graph with "loading data" before firing the request for the measurements
- [x] use `requestAnimationFrame` to avoid Highcharts not finding the div
- [x] when hovering on graph the bubble with the details of each measurements should be on top and caret should be pointing down
- [x] use correct colors in graph and not greys
- [x] redraw on heat map change
- [x] avoid the double draw caused by ToggleSession and ToggleSessionFromAngular

Review commit by commit if you prefer to check smaller changes

### CODE REVIEW

File: app/javascript/angular/code/controllers/_sessions_list_ctrl.js [hunk 1]
```diff
@@ -110,23 +110,26 @@ export const SessionsListCtrl = (
...
+    $scope.toggleSession(
+      data.session_id,
+      elmApp.ports.toggleSessionSelection.send
+    );
     $scope.$apply();
   });
 
-  $scope.toggleSession = function(sessionId, markerSelected) {
+  $scope.toggleSession = function(sessionId, callback) {
     if (this.isSessionDisabled(sessionId)) {
```
Comment (AkkarinForest) [inline app/javascript/angular/code/controllers/_sessions_list_ctrl.js]: do we still need that if we have radio buttons now?

Comment (3v0k4) [reply app/javascript/angular/code/controllers/_sessions_list_ctrl.js]: Good catch. Prolly not but will leave this to another round of removals

File: app/javascript/elm/src/Main.elm [hunk 2]
```diff
@@ -317,11 +317,18 @@ update msg model =
...
+                        ( { model | selectedSession = NotAsked }, Cmd.none )
+
+                    else
+                        ( model, SelectedSession.fetch model.sensors model.selectedSensorId model.page id (RemoteData.fromResult >> GotSession) )
+
+                ( _, Just id ) ->
                     ( model, SelectedSession.fetch model.sensors model.selectedSensorId model.page id (RemoteData.fromResult >> GotSession) )
 
-                Nothing ->
+                ( _, Nothing ) ->
```
[the remark this answers is not in the document]
Comment (3v0k4) [reply app/javascript/elm/src/Main.elm]: I see what you mean. At the same time, I'm sure sure how to simplify that w/o making it more complex.

I'm not sure it's an impossible state. I think that from Angular we could expect anything coming at this point.

**Comment by 3v0k4** [comment]
@AkkarinForest thanks for the CR!

File: app/javascript/packs/elm.js [hunk 3]
```diff
@@ -140,7 +140,7 @@ const setupHeatMap = () => {
 };
 
 const draw = fnc => ({ times, streamId, heat, sensor }) =>
-  fnc({ sensor, heat, times, streamId });
+  window.requestAnimationFrame(() => fnc({ sensor, heat, times, streamId }));
```
Comment (gitjul) [inline app/javascript/packs/elm.js]: interesting, didn't know that trick...

File: app/javascript/elm/src/Main.elm [hunk 4]
```diff
@@ -448,10 +448,10 @@ toGraphParams thresholds selectedSession sensors selectedSensorId =
...
 
         levels =
-            [ { from = threshold1, to = threshold2, color = "#aaa" }
-            , { from = threshold2, to = threshold3, color = "#bbb" }
-            , { from = threshold3, to = threshold4, color = "#ccc" }
-            , { from = threshold4, to = threshold5, color = "#ddd" }
+            [ { from = threshold1, to = threshold2, className = "first-band" }
+            , { from = threshold2, to = threshold3, className = "second-band" }
+            , { from = threshold3, to = threshold4, className = "third-band" }
+            , { from = threshold4, to = threshold5, className = "fourth-band" }
```
Comment (gitjul) [inline app/javascript/elm/src/Main.elm]: why `-band`? (just trying to understand the meaning, what it reflects)

Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: (in case I wanna change sth here as you suggested)

Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: another words: what's a graph band?

File: app/javascript/packs/elm.js [hunk 5]
```diff
@@ -126,11 +126,9 @@ const setupHeatMap = () => {
       console.warn("heatmap updated", Object.values(toValues(node.noUiSlider)));
 
       // changing extremes could have changed middle values
```
Comment (gitjul) [inline app/javascript/packs/elm.js]: is this comment still relevant?

File: app/javascript/elm/src/Main.elm (hunk 4 above)
Comment (3v0k4) [reply app/javascript/elm/src/Main.elm]: I guess this should answer the question https://www.highcharts.com/docs/chart-concepts/plot-bands-and-plot-lines

File: app/javascript/packs/elm.js (hunk 5 above)
Comment (3v0k4) [reply app/javascript/packs/elm.js]: yep

**Comment by 3v0k4** [comment]
Thanks for the CR @gitjul

**Pull request merged**